Jenkinsfile

```diff
@@ -8,118 +8,554 @@ retriever: modernSCM(
+                }
+              }
+            }
+          }
+        }
+        stage('Auditbeat'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            anyOf {
+              changeset(glob: 'libbeat/**')
+              changeset(glob: 'auditbeat/**')
+              expression { return params.runAllStages }
+            }
+          }
+          stages {
+            stage('Auditbeat oss'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Auditbeat oss Linux", "-C auditbeat testsuite")
+                }
+              }
+            }
+            stage('Auditbeat crosscompile'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Auditbeat oss crosscompile", "-C auditbeat crosscompile")
+                }
+              }
+            }
+            stage('Auditbeat x-pack'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Auditbeat x-pack Linux", "-C x-pack/auditbeat testsuite")
+                }
+              }
+            }
+            stage('Auditbeat Mac OS X'){
+              agent { label 'macosx' }
+              options { skipDefaultCheckout() }
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Auditbeat oss Mac OS X", "TEST_ENVIRONMENT=0 -C auditbeat testsuite")
+                }
+              }
+            }
+          }
+        }
+        stage('Libbeat'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            anyOf {
+              changeset(glob: 'libbeat/**')
+              expression { return params.runAllStages }
+            }
+          }
+          stages {
+            stage('Libbeat oss'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat oss Linux", "-C libbeat testsuite")
+                }
+              }
+            }
+            stage('Libbeat crosscompile'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat oss crosscompile", "-C libbeat crosscompile")
+                }
+              }
+            }
+            stage('Libbeat stress-tests'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat stress-tests", "STRESS_TEST_OPTIONS='-timeout=20m -race -v -parallel 1' -C libbeat stress-tests")
+                }
+              }
+            }
+            stage('Libbeat x-pack'){
+              steps {
+                withBeatsEnv(){
+                  makeTarget("Libbeat x-pack Linux", "-C x-pack/libbeat testsuite")
+                }
+              }
+            }
+          }
+        }
+        stage('Metricbeat Unit tests'){
```

Comment:
How come the Metricbeat stages are not grouped under common "Metricbeat" stage? Is that so they can execute in parallel on different agents?

Reply:
yep, it is for run those stages in parallel, they are the slowest